plugins/inputs/ravendb/ravendb.go

```diff
@@ -0,0 +1,680 @@
+			fields["time_since_last_backup_in_sec"] = dbResponse.TimeSinceLastBackupInSec
+		}
+
+		acc.AddFields("ravendb_databases", fields, tags)
+	}
+}
+
+func gatherIndexes(r *RavenDB, acc telegraf.Accumulator) {
+	if !r.GatherIndexStats {
+		return
+	}
+
+	indexesResponse := &indexesMetricResponse{}
+
+	err := r.requestJSON("/admin/monitoring/v1/indexes"+prepareDbNamesUrlPart(r.IndexStatsDbs), &indexesResponse)
+	if err != nil {
+		acc.AddError(err)
+		return
+	}
+
+	for _, indexResponse := range indexesResponse.Results {
+		tags := map[string]string{"url": r.URL}
+		tags["database_name"] = indexResponse.DatabaseName
+		tags["index_name"] = indexResponse.IndexName
+		tags["node_tag"] = indexesResponse.NodeTag
+
+		if indexesResponse.PublicServerUrl != "" {
+			tags["public_server_url"] = indexesResponse.PublicServerUrl
+		}
+
+		fields := map[string]interface{}{
+			"priority":        indexResponse.Priority,
+			"state":           indexResponse.State,
+			"errors":          indexResponse.Errors,
+			"lock_mode":       indexResponse.LockMode,
+			"is_invalid":      boolToInt(indexResponse.IsInvalid),
+			"status":          indexResponse.Status,
+			"mapped_per_sec":  indexResponse.MappedPerSec,
+			"reduced_per_sec": indexResponse.ReducedPerSec,
+			"type":            indexResponse.Type,
+		}
+
+		if indexResponse.TimeSinceLastQueryInSec != nil {
+			fields["time_since_last_query_in_sec"] = indexResponse.TimeSinceLastQueryInSec
+		}
+
+		if indexResponse.TimeSinceLastIndexingInSec != nil {
+			fields["time_since_last_indexing_in_sec"] = indexResponse.TimeSinceLastIndexingInSec
+		}
+
+		acc.AddFields("ravendb_indexes", fields, tags)
+	}
+}
+
+func gatherCollections(r *RavenDB, acc telegraf.Accumulator) {
+	if !r.GatherCollectionStats {
+		return
+	}
+
+	collectionsResponse := &collectionsMetricResponse{}
+
+	err := r.requestJSON("/admin/monitoring/v1/collections"+prepareDbNamesUrlPart(r.IndexStatsDbs), &collectionsResponse)
+	if err != nil {
+		acc.AddError(err)
+		return
+	}
+
+	for _, collectionMetrics := range collectionsResponse.Results {
+		tags := map[string]string{"url": r.URL}
+		tags["node_tag"] = collectionsResponse.NodeTag
+		tags["database_name"] = collectionMetrics.DatabaseName
+		tags["collection_name"] = collectionMetrics.CollectionName
+
+		if collectionsResponse.PublicServerUrl != "" {
+			tags["public_server_url"] = collectionsResponse.PublicServerUrl
+		}
+
+		fields := map[string]interface{}{
+			"documents_count":          collectionMetrics.DocumentsCount,
+			"total_size_in_bytes":      collectionMetrics.TotalSizeInBytes,
+			"documents_size_in_bytes":  collectionMetrics.DocumentsSizeInBytes,
+			"tombstones_size_in_bytes": collectionMetrics.TombstonesSizeInBytes,
+			"revisions_size_in_bytes":  collectionMetrics.RevisionsSizeInBytes,
+		}
+
+		acc.AddFields("ravendb_collections", fields, tags)
+	}
+}
+
+func prepareDbNamesUrlPart(dbNames []string) string {
+	if len(dbNames) == 0 {
+		return ""
+	}
+
```

Comment:
This is the result anyway, so you can skip this test.

Reply:
removed

Reply:
Thx, please mark as resolved.